Automattic/woocommerce-payments
# Ensure shortcode renders for Privacy Policy & Terms of Service

Fixes #7600

#### Changes proposed in this Pull Request









This PR allows the following shortcodes to be rendered in the WooPay Checkout preview: `[privacy_policy]` and `[terms]`. 

It should also be noted that the shortcodes have been renamed from `[privacy_policy_link]` to `[privacy_policy]` and `[terms_of_service_link]` to `[terms]`. These changes were made in order to have a consistent naming across WooPayments and WooCommerce.

https://github.com/Automattic/woocommerce-payments/assets/6844516/454f8673-557d-42f7-9bcf-37bc5b22533c


#### Testing instructions





**Test: Ensure the Privacy Policy and Terms of Service links appear in WooPay checkout**
* As a customer, navigate to the merchant shop, add an item to the cart, and proceed to WooPay checkout.
* Ensure that the **Policies and custom test** appear in the WooPay checkout page. 
* Ensure the Privacy Policy and Terms of Service links work as intended.

**Test: Ensure shortcode is rendered to look like a link, in the preview**
* As a merchant, ensure a Privacy Policy and Terms of Service page are linked.
* Navigate to **WooCommerce > Settings > Payments > WooPayments > WooPay > Checkout appearance**.
* Update the **Policies and custom text** so that it includes the following shortcodes: `[privacy_policy]` and `[terms]`.
* Ensure the shortcodes are rendered and appear as links. Also, ensure that they are not interactive/clickable.

**Regression test: Ensure other elements are not rendered in the preview**
* Update the **Policies and custom text** so that it includes other shortcodes, HTML elements, JS script, etc. 
* Ensure these other elements are not rendered.


-------------------


**Post merge**
